File: Medium/2026-04-23-2615-Sum-of-Distances/solution.py
class Solution:
    def distance(self, nums: List[int]) -> List[int]:
        from collections import defaultdict
        
        index_map = defaultdict(list)
        n = len(nums)
        arr = [0] * n
        
        for i in range(n):
            index_map[nums[i]].append(i)
        
        for indices in index_map.values():
            if len(indices) < 2:
                continue
            
            total = sum(indices)
            count = len(indices)
            for i in range(count):
                left_count = i
                right_count = count - i - 1
                left_sum = indices[i] * left_count - sum(indices[:i])
                right_sum = sum(indices[i + 1:]) - indices[i] * right_count
                arr[indices[i]] = left_sum + right_sum
        
        return arr

File: Medium/2026-04-23-2615-Sum-of-Distances/solution.js
var distance = function(nums) {
    const n = nums.length;
    const map = new Map();
    const result = new Array(n).fill(0);
    
    for (let i = 0; i < n; i++) {
        if (!map.has(nums[i])) {
            map.set(nums[i], []);
        }
        map.get(nums[i]).push(i);
    }
    
    for (const indices of map.values()) {
        const m = indices.length;
        if (m < 2) continue;
        
        const prefixSum = new Array(m + 1).fill(0);
        for (let i = 1; i <= m; i++) {
            prefixSum[i] = prefixSum[i - 1] + indices[i - 1];
        }
        
        for (let i = 0; i < m; i++) {
            const leftCount = i;
            const rightCount = m - 1 - i;
            const leftSum = prefixSum[i];
            const rightSum = prefixSum[m] - prefixSum[i + 1];
            result[indices[i]] = leftCount * indices[i] - leftSum + rightSum - rightCount * indices[i];
        }
    }
    
    return result;
};